eslint.config.base.js

```diff
@@ -0,0 +1,1266 @@
+import fs from 'node:fs';
+import path from 'node:path';
+
+import stylisticPlugin from '@stylistic/eslint-plugin-js';
+import vitestPlugin from '@vitest/eslint-plugin';
+import restrictedGlobals from 'confusing-browser-globals';
+import importPlugin from 'eslint-plugin-import';
+import jsxA11yPlugin from 'eslint-plugin-jsx-a11y';
+import promisePlugin from 'eslint-plugin-promise';
+import reactPlugin from 'eslint-plugin-react';
+import reactHooksPlugin from 'eslint-plugin-react-hooks';
+import regexpPlugin from 'eslint-plugin-regexp';
+import simpleImportSortPlugin from 'eslint-plugin-simple-import-sort';
+import storybookPlugin from 'eslint-plugin-storybook';
+import testingLibraryPlugin from 'eslint-plugin-testing-library';
+import unicornPlugin from 'eslint-plugin-unicorn';
+import globals from 'globals';
+import _ from 'lodash';
+import tseslint from 'typescript-eslint';
+
+const DEFAULT_OPTS = {
+  typescript: false,
+  react: false,
+  storybook: false,
+  vitest: false,
+  cypress: false,
+  reactTestingLibrary: false,
+  cypressTestingLibrary: false,
+};
+
+export function detectOpts(projectDir) {
+  const pkgPath = path.resolve(projectDir, './package.json');
+  const tsconfigPath = path.resolve(projectDir, './tsconfig.json');
+
+  const pkg = JSON.parse(fs.readFileSync(pkgPath));
+  const deps = new Set([
+    ...Object.keys(pkg.dependencies || {}),
+    ...Object.keys(pkg.devDependencies || {}),
+  ]);
+
+  return {
+    typescript: deps.has('typescript') && fs.existsSync(tsconfigPath),
+    react: deps.has('react'),
+    storybook: deps.has('storybook'),
+    vitest: deps.has('vitest'),
+    cypress: deps.has('cypress'),
+    reactTestingLibrary: deps.has('@testing-library/react'),
+    cypressTestingLibrary: deps.has('@testing-library/cypress'),
+  };
+}
+
+// eslint-disable-next-line complexity
+export function createConfig(opts = {}) {
+  const {
+    typescript: withTypeScript,
+    react: withReact,
+    storybook: withStorybook,
+    vitest: withVitest,
+    cypress: withCypress,
+    reactTestingLibrary: withReactTestingLibrary,
+    cypressTestingLibrary: withCypressTestingLibrary,
+  } = { ...DEFAULT_OPTS, ...opts };
+
+  return tseslint.config(
+    _.compact([
```

Comment:
lodash's `compact` utility allows passing `withTypeScript && { ... }` into the array instead of `...(withTypeScript ? [{ ... }] : [])`.